Sources/SnapshotTesting/Common/ForwardingWKNavigationDelegate.swift

```diff
@@ -0,0 +1,133 @@
+#if os(macOS)
+import Cocoa
+#endif
+#if os(iOS)
+import UIKit
+#endif
+
+#if os(iOS) || os(macOS)
+import WebKit
+
+final class ForwardingWKNavigationDelegate: NSObject {
+  var didFinish: (() -> Void)?
+  var original: WKNavigationDelegate?
+  init(didFinish: @escaping () -> Void = {}, originalDelegate: WKNavigationDelegate?) {
+    self.original = originalDelegate
+    self.didFinish = didFinish
+  }
+
+  override public func responds(to aSelector: Selector!) -> Bool {
+    if decidePolicyPre13 == aSelector, original?.responds(to: aSelector) != true {
+      return NSObject.responds(to: aSelector)
+    }
+
+    if
+      #available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *),
+      decidePolicy13 == aSelector,
+      original?.responds(to: aSelector) != true
+    {
+      return NSObject.responds(to: aSelector)
+    }
+
+    return interceptedSelectors.contains(aSelector)
+      || original?.responds(to: aSelector) == true
+      || NSObject.responds(to: aSelector)
+  }
+
+  override public func forwardingTarget(for aSelector: Selector!) -> Any? {
+    interceptedSelectors.contains(aSelector)
+      ? nil
+      : original
+  }
+
+  func finish() {
+    didFinish?()
+    didFinish = nil
+  }
+}
+
+extension ForwardingWKNavigationDelegate: WKNavigationDelegate {
+  @available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *)
+  func webView(
+    _ webView: WKWebView,
+    decidePolicyFor navigationAction: WKNavigationAction,
+    preferences: WKWebpagePreferences,
+    decisionHandler: @escaping (WKNavigationActionPolicy, WKWebpagePreferences) -> Void
+  ) {
+    original?.webView?(
+      webView,
+      decidePolicyFor: navigationAction,
+      preferences: preferences,
+      decisionHandler: { [weak self] policy, prefs in
+        decisionHandler(policy, prefs)
+        if policy == .cancel {
+          self?.finish()
```

Comment:
If the user did choose to implement this method, we need to check if the `decisionHandler` is called with `.cancel`. If that is the case we need to call `finish` because `webView(_:didFinish)` or  webView(_:didFail:withError)` won't be called by the webView!